server/lib/chat/chat-service.test.ts

```diff
@@ -0,0 +1,1505 @@
+    id: makeSbChannelId(1),
+    name: 'ShieldBattery',
+    private: false,
+    official: true,
+  }
+  const testChannel: ChannelInfo = {
+    id: makeSbChannelId(2),
+    name: 'test',
+    private: false,
+    official: false,
+  }
+
+  const user1: SbUser = { id: makeSbUserId(1), name: 'USER_NAME_1' }
+  const user2: SbUser = { id: makeSbUserId(2), name: 'USER_NAME_2' }
+
+  const userPermissions: SbPermissions = {
+    editPermissions: false,
+    debug: false,
+    banUsers: false,
+    manageLeagues: false,
+    manageMaps: false,
+    manageMapPools: false,
+    manageMatchmakingSeasons: false,
+    manageMatchmakingTimes: false,
+    manageRallyPointServers: false,
+    massDeleteMaps: false,
+    moderateChatChannels: false,
+  }
+  const channelPermissions: ChannelPermissions = {
+    kick: false,
+    ban: false,
+    changeTopic: false,
+    togglePrivate: false,
+    editPermissions: false,
+  }
+  const user1ShieldBatteryChannelEntry: UserChannelEntry = {
+    userId: user1.id,
+    channelId: shieldBatteryChannel.id,
+    joinDate: new Date('2023-03-11T00:00:00.000Z'),
+    channelPermissions,
+  }
+  const user1TestChannelEntry: UserChannelEntry = {
+    userId: user1.id,
+    channelId: testChannel.id,
+    joinDate: new Date('2023-03-11T00:00:00.000Z'),
+    channelPermissions,
+  }
+  const user2ShieldBatteryChannelEntry: UserChannelEntry = {
+    userId: user2.id,
+    channelId: shieldBatteryChannel.id,
+    joinDate: new Date('2023-03-11T00:00:00.000Z'),
+    channelPermissions,
+  }
+  const user2TestChannelEntry: UserChannelEntry = {
+    userId: user2.id,
+    channelId: testChannel.id,
+    joinDate: new Date('2023-03-11T00:00:00.000Z'),
+    channelPermissions,
+  }
+
+  const joinUser1ShieldBatteryChannelMessage: ChatMessage = {
+    msgId: 'MESSAGE_ID',
+    userId: user1.id,
+    userName: user1.name,
+    channelId: shieldBatteryChannel.id,
+    sent: new Date('2023-03-11T00:00:00.000Z'),
+    data: {
+      type: ServerChatMessageType.JoinChannel,
+    },
+  }
+  const joinUser1TestChannelMessage: ChatMessage = {
+    msgId: 'MESSAGE_ID',
+    userId: user1.id,
+    userName: user1.name,
+    channelId: testChannel.id,
+    sent: new Date('2023-03-11T00:00:00.000Z'),
+    data: {
+      type: ServerChatMessageType.JoinChannel,
+    },
+  }
+  const joinUser2ShieldBatteryChannelMessage: ChatMessage = {
+    msgId: 'MESSAGE_ID',
+    userId: user2.id,
+    userName: user2.name,
+    channelId: shieldBatteryChannel.id,
+    sent: new Date('2023-03-11T00:00:00.000Z'),
+    data: {
+      type: ServerChatMessageType.JoinChannel,
+    },
+  }
+  const joinUser2TestChannelMessage: ChatMessage = {
+    msgId: 'MESSAGE_ID',
+    userId: user2.id,
+    userName: user2.name,
+    channelId: testChannel.id,
+    sent: new Date('2023-03-11T00:00:00.000Z'),
+    data: {
+      type: ServerChatMessageType.JoinChannel,
+    },
+  }
+
+  // A helper method that mocks everything correctly to "join" a user to a channel.
```

Comment:
nit: If you're documenting a method, use a doc comment

Reply:
other nit: this can just be a normal function, there is no need for it to be an arrow function